src/main/java/com/teragrep/cfe_39/consumers/kafka/DatabaseOutput.java

```diff
@@ -0,0 +1,396 @@
+    @Override
+    public void accept(List<RecordOffsetObject> recordOffsetObjectList) {
+        long thisTime = Instant.now().toEpochMilli();
+        long ftook = thisTime - lastTimeCalled;
+        topicCounter.setKafkaLatency(ftook);
+        LOGGER.debug(ANSI_BLUE + "Fuura searching your batch for <[" + table + "]> with records <" + recordOffsetObjectList.size() + "> and took  <" + (ftook) + "> milliseconds. <" + (recordOffsetObjectList.size() * 1000L / ftook) + "> EPS. " + ANSI_RESET);
+        long batchBytes = 0L;
+
+        //  The recordOffsetObjectList loop will go through all the objects in the list.
+        //  While it goes through the list, the contents of the objects are serialized into an AVRO-file.
+        //  When the file size is about to go above 64M, commit the file into HDFS using the latest topic/partition/offset values as the filename and start fresh with a new empty AVRO-file.
+        //  Serialize the object that was going to make the file go above 64M into the now empty AVRO-file and continue the loop.
+        // TODO: If the prod-environment recordOffsetObjectList ordering is different from what it is in the test environment, add a function that reorders the list based on partition and offset (or better yet, make several AVRO-files that are being used at the same time rather than doing it one AVRO-file at a time as the offset ordering within partitions should always be correct in all scenarios).
+        RecordOffsetObject lastObject = null;
+        long start = Instant.now().toEpochMilli(); // Starts measuring performance here. Measures how long it takes to process the whole recordOffsetObjectList.
+        // This loop goes through all the records of the mock data in a single session.
+        for (RecordOffsetObject recordOffsetObject : recordOffsetObjectList) {
+            // Initializing syslogAvroWriter.
+            if (syslogAvroWriter == null) {
+                try {
+                    writableQueue.setQueueNamePrefix(recordOffsetObject.topic+recordOffsetObject.partition);
+                    syslogFile =
+                            writableQueue.getNextWritableFile();
+                    //  The HDFS filename is only finalized when the AVRO-serialized file is finalized, because every Kafka-record added to the file is going to change the offset that is going to be used for the filename.
+                    syslogAvroWriter = new SyslogAvroWriter(syslogFile);
+                    approximatedSize = syslogAvroWriter.getFileSize();  // resets the size approximation.
+                    lastObject = recordOffsetObject;
+                } catch (IOException ioException) {
+                    throw new IllegalArgumentException(ioException);
+                }
+            } else {
+                try {
+                    // check if new partition
+                    assert lastObject != null;
```

Comment:
If this check fails, it will throw an AssertionError, which will stop execution without a possibility to recover. Disregard this if that's intended.